test/uk/gov/hmrc/tai/service/IncomeServiceSpec.scala

```diff
@@ -139,6 +139,50 @@ class IncomeServiceSpec extends BaseSpec {
       val result = sut.taxCodeIncomes(nino, TaxYear())(HeaderCarrier()).futureValue
 
 
+      result mustBe taxCodeIncomes.map(_.copy(status = Ceased))
+    }
+
+
+    "return the list of taxCodeIncomes for passed nino even if employments is nill" in {
+      val taxCodeIncomes = Seq(
+        TaxCodeIncome(
+          EmploymentIncome,
+          Some(1),
+          BigDecimal(0),
+          "EmploymentIncome",
+          "1150L",
+          "Employer1",
+          Week1Month1BasisOperation,
+          Live,
+          BigDecimal(0),
+          BigDecimal(0),
+          BigDecimal(0)
+        ),
+        TaxCodeIncome(
+          EmploymentIncome,
+          Some(2),
+          BigDecimal(0),
+          "EmploymentIncome",
+          "1100L",
+          "Employer2",
+          OtherBasisOperation,
+          Live,
+          BigDecimal(0),
+          BigDecimal(0),
+          BigDecimal(0))
+      )
+
+      val mockEmploymentService = mock[EmploymentService]
+      val mockIncomeRepository = mock[IncomeRepository]
+      when(mockIncomeRepository.taxCodeIncomes(any(), any())(any()))
+        .thenReturn(Future.successful(taxCodeIncomes))
+
+      when(mockEmploymentService.employments(any[Nino], any[TaxYear])(any[HeaderCarrier]))
+        .thenReturn(Future.successful(Seq.empty))
+
+      val sut = createSUT(employmentService = mockEmploymentService, incomeRepository = mockIncomeRepository)
+      val result = sut.taxCodeIncomes(nino, TaxYear())(HeaderCarrier()).futureValue
+
       result mustBe taxCodeIncomes
```

Comment:
If Im thinking right, if we cant get the status from employments then we get it from the taxCodeIncome? Should we check that the status is Live here aswell to see it is coming from TaxCodeIncome?

Reply:
Something like this?

`result mustBe taxCodeIncomes.map(_.copy(status = Live))`

Reply:
yep agreed, my bad